url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/DATA/Filler.java

```diff
@@ -0,0 +1,980 @@
+		this.teachers = ds.getTeachers();
+		this.types = ds.getTypes();
+		this.MWWH = ds.getMWWH();
+		this.ps = aPs == null ? System.out : aPs;
+		
+	}
+	
+	/**
+	 * Dans l'ordre spécifié par la paramètre,
+	 * cette méthode devra attribuer les Constrainables
+	 * jusqu'à ce que l'emploi du temps soit complet.
+	 * Le paramètre mode détermine le comportement de la méthode handleError en cas d'échec.
+	 * @param order l'ordre général dans lequel traiter les données (déterminé par la méthode computeConstraints()).
+	 * @param mode Le comportement à appliquer si l'une des méthodes takeCareOf([...]) renvoie false ( <=> rencontre une erreur)
+	 * @return le nombre d'erreurs rencontrées lors du remplissage.
+	 */
+	public int fill(HashMap<Constrainable, Double> order, int mode)	{
+		
+		this.mode = mode;
+		ArrayList<Group> groupsOrder = new ArrayList<Group>();
+		ArrayList<Field> fieldsOrder = new ArrayList<Field>();
+		
+		this.steps = new HashMap<Integer, Integer>();
+		this.errorsIndexes = new HashMap<Integer, Integer>();
+		this.done = new ArrayList<Lesson>();
+		
+		for (Object o : order)	{
+			Constrainable c = (Constrainable) o;
+			if (c instanceof Group)
+				groupsOrder.add((Group)c);
+			
+			else if (c instanceof Field)
+				fieldsOrder.add((Field)c);
+		}
+		
+		int errors = 0;
+		
+		//ps.println("Filler.fill() : " + order);
+		
+		for (int i = 0 ; i < order.size() ; i++)	{
+
+			steps.put(i, done.size());
+			//ps.println("Filler.fill().steps : " + steps + "\n\t\t\t" + done.size());
+			Constrainable c = order.getKey(i);
+			//ps.println("Filler.fill() #steps : " + i + " " + steps.values());// + "\n\n" + order);
+			//ps.println("Filler.fill : " + i + " " + c);
+			
+			if (c instanceof Field)	{
+
+				if (!this.takeCareOf((Field) c, groupsOrder))	{
+					++errors;
+					ps.println("Filler.fill#Error : " + c);
+					i = this.handleError(i, order);
+				}
+			}
+			
+			else if (c instanceof ClassType)	{
+
+				if (!this.takeCareOf((ClassType) c, groupsOrder, fieldsOrder))	{
+					++errors;
+					ps.println("Filler.fill#Error : " + c);
+					i = this.handleError(i, order);
+				}
+			}
+			
+			else if (c instanceof Teacher)	{
+
+				if (!this.takeCareOf((Teacher) c, fieldsOrder, groupsOrder))	{
+					++errors;
+					ps.println("Filler.fill#Error : " + c);
+					i = this.handleError(i, order);
+				}
+			}
+			
+			else if (c instanceof Group)	{
+				
+				if (!this.takeCareOf((Group) c, fieldsOrder))	{
+					++errors;
+					ps.println("Filler.fill#Error : " + c);
+					i = this.handleError(i, order);
+				}
+			}
+		}
+		return errors;
+	}
+	
+	/**
+	 * Attribue les profs aux groupes dans l'ordre indiqué par le paramètre order (par ordre de contrainte dans l'idée)
+	 * @param order l'ordre dans lequel traiter les matières lors de l'attribution.
+	 * @return false si une erreur a été rencontrée (le descriptif de l'erreur est détailler dans la sortie terminal).
+	 */
+	public boolean attributeTeachers(ArrayList<Field> order)	{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 189-219 into a separate method.